File: DevTests/Addons/_Addons/TestStepsModule/TestSteps.js
"use strict";
/* Copyright © 2020 Softel vdm, Inc. - https://yetawf.com/Documentation/YetaWF/DevTests#License */
var YetaWF_DevTests;
(function (YetaWF_DevTests) {
    var TestStepsModule = /** @class */ (function () {
        function TestStepsModule(id) {
            this.Module = $YetaWF.getElementById(id);
            $YetaWF.registerEventHandler(this.Module, "click", "input[name='Clear']", function (ev) {
                if (typeof YetaWF_Panels !== "undefined" && YetaWF_Panels.StepInfoComponent) {
                    YetaWF_Panels.StepInfoComponent.clearAll(); // deactivate all steps
                }
                return true;
            });
            $YetaWF.registerEventHandler(this.Module, "click", "input[name='ActivateAll']", function (ev) {
                if (typeof YetaWF_Panels !== "undefined" && YetaWF_Panels.StepInfoComponent) {
                    YetaWF_Panels.StepInfoComponent.setAllActive(["YetaWF_DevTests_Step"]); // activates all named steps that start with the specified name(s)
                }
                return true;
            });
            $YetaWF.registerEventHandler(this.Module, "click", "input[name='ActivateLast']", function (ev) {
                if (typeof YetaWF_Panels !== "undefined" && YetaWF_Panels.StepInfoComponent) {
                    YetaWF_Panels.StepInfoComponent.setLastActive(["YetaWF_DevTests_Step"]); // activates the last named step that start with the specified name(s)
                }
                return true;
            });
            $YetaWF.registerEventHandler(this.Module, "click", "input[name='Step1']", function (ev) {
                if (typeof YetaWF_Panels !== "undefined" && YetaWF_Panels.StepInfoComponent) {
                    YetaWF_Panels.StepInfoComponent.setActive({ Name: "YetaWF_DevTests_Step1" });
                }
                return true;
            });
            $YetaWF.registerEventHandler(this.Module, "click", "input[name='Step2']", function (ev) {
                if (typeof YetaWF_Panels !== "undefined" && YetaWF_Panels.StepInfoComponent) {
                    YetaWF_Panels.StepInfoComponent.setActive({ Name: "YetaWF_DevTests_Step2" });
                }
                return true;
            });
            $YetaWF.registerEventHandler(this.Module, "click", "input[name='Step3']", function (ev) {
                if (typeof YetaWF_Panels !== "undefined" && YetaWF_Panels.StepInfoComponent) {
                    YetaWF_Panels.StepInfoComponent.setActive({ Name: "YetaWF_DevTests_Step3" });
                }
                return true;
            });
            $YetaWF.registerEventHandler(this.Module, "click", "input[name='Step4']", function (ev) {
                if (typeof YetaWF_Panels !== "undefined" && YetaWF_Panels.StepInfoComponent) {
                    YetaWF_Panels.StepInfoComponent.setActive({ Name: "YetaWF_DevTests_Step4" });
                }
                return true;
            });
            $YetaWF.registerEventHandler(this.Module, "click", "input[name='Step12']", function (ev) {
                if (typeof YetaWF_Panels !== "undefined" && YetaWF_Panels.StepInfoComponent) {
                    YetaWF_Panels.StepInfoComponent.setActive({ Name: "YetaWF_DevTests_Step2" });
                    YetaWF_Panels.StepInfoComponent.setEnabled([{ Name: "YetaWF_DevTests_Step1", Url: "/" }]);
                }
                return true;
            });
            $YetaWF.registerEventHandler(this.Module, "click", "input[name='Step13']", function (ev) {
                if (typeof YetaWF_Panels !== "undefined" && YetaWF_Panels.StepInfoComponent) {
                    YetaWF_Panels.StepInfoComponent.setActive({ Name: "YetaWF_DevTests_Step3" });
                    YetaWF_Panels.StepInfoComponent.setEnabled([{ Name: "YetaWF_DevTests_Step1", Url: "/" }, { Name: "YetaWF_DevTests_Step2", Url: "/" }]);
                }
                return true;
            });
            $YetaWF.registerEventHandler(this.Module, "click", "input[name='Step14']", function (ev) {
                if (typeof YetaWF_Panels !== "undefined" && YetaWF_Panels.StepInfoComponent) {
                    YetaWF_Panels.StepInfoComponent.setActive({ Name: "YetaWF_DevTests_Step4" });
                    YetaWF_Panels.StepInfoComponent.setEnabled([{ Name: "YetaWF_DevTests_Step1", Url: "/" }, { Name: "YetaWF_DevTests_Step2", Url: "/" }, { Name: "YetaWF_DevTests_Step3", Url: "/" }]);
                }
                return true;
            });
        }
        return TestStepsModule;
    }());
    YetaWF_DevTests.TestStepsModule = TestStepsModule;
})(YetaWF_DevTests || (YetaWF_DevTests = {}));

File: DevTests/Views/HTML/TestSteps.cs
/* Copyright © 2021 Softel vdm, Inc. - https://yetawf.com/Documentation/YetaWF/DevTests#License */

using System.Threading.Tasks;
using YetaWF.Core.Components;
using YetaWF.Core.Packages;
using YetaWF.Core.Support;
using YetaWF.Modules.ComponentsHTML.Components;
using YetaWF.Modules.DevTests.Controllers;
using YetaWF.Modules.DevTests.Modules;

namespace YetaWF.Modules.DevTests.Views {

    public class TestStepsView : YetaWFView, IYetaWFView<TestStepsModule, TestStepsModuleController.Model> {

        public const string ViewName = "TestSteps";

        public override Package GetPackage() { return AreaRegistration.CurrentPackage; }
        public override string GetViewName() { return ViewName; }

        public async Task<string> RenderViewAsync(TestStepsModule module, TestStepsModuleController.Model model) {

            HtmlBuilder hb = new HtmlBuilder();

            hb.Append($@"
{await HtmlHelper.ForDisplayContainerAsync(model, "PropertyList")}

<input type='button' value='clear()' name='Clear' />
<input type='button' value='activateAll()' name='ActivateAll' />
<input type='button' value='activateLast()' name='ActivateLast' />
<br/>
<br/>
<input type='button' value='step1' name='Step1' />
<input type='button' value='step2' name='Step2' />
<input type='button' value='step3' name='Step3' />
<input type='button' value='step4' name='Step4' />
<br/>
<br/>
<input type='button' value='step1' name='Step1' />
<input type='button' value='step1-step2' name='Step12' />
<input type='button' value='step1-step3' name='Step13' />
<input type='button' value='step1-step4' name='Step14' />");

            Manager.ScriptManager.AddLast($@"
if (typeof YetaWF_Panels !== 'undefined' && YetaWF_Panels.StepInfoComponent) {{
    YetaWF_Panels.StepInfoComponent.setActive({{Name: 'YetaWF_DevTests_Step1'}});
}}");

            return hb.ToString();
        }
    }
}
